### DISCUSSION TITLE
How to generate a 2D diamond lattice network?

### DISCUSSION THREAD

**Discussion by kunal0kumar0sinha** [PMEAL/OpenPNM#2840] [FAQ]
I have been trying to generate a diamond grid (attached figure). Can you please suggest how to do this?

![image](https://github.com/PMEAL/OpenPNM/assets/123363487/e9bf344a-33bc-4133-9c86-3a5c05b5e23a)

**Comment by jgostick** [comment]
I'm sure I had a better way to do this at some point, but this is what I came up with just now:

```python

import openpnm as op
pn = op.network.BodyCenteredCubic([10, 10, 2])
op.topotools.trim(pn, pores=pn.pores('zmax'))
op.topotools.trim(pn, throats=pn.throats('body_to_body'))
op.topotools.trim(pn, throats=pn.throats('corner_to_corner'))
pn['pore.coords'] *= [1, 1, 0]
del pn.params['dimensionality']
h = op.visualization.plot_connections(pn)
op.visualization.plot_coordinates(pn, ax=h)

```

![Figure_1](https://github.com/PMEAL/OpenPNM/assets/5092049/dd36b01b-9c76-4795-8c21-50222a6b443a)

**Reply by kunal0kumar0sinha** [reply]
Thank you.

**Comment by jgostick** [comment]
I think this is the other way that I've done this in that past:

``` python
from skimage.morphology import diamond
im = diamond(20)
pn = op.network.CubicTemplate(im)
pn = op.topotools.rotate_coords(pn, a=0, b=0, c=45)
```

Perhaps this is a bit more logical.
